File: backend/src/book/dto/book.dto.ts
import { UUID } from "crypto";
import e from "express";
import { Book } from "../entities/book.entity";

export class BookDto {
    private id: UUID;
    private isbn: string;
    private category: string;
    private title: string;
    private author: string;
    private language: string;
    private publicationYear: number;
    private publisher: string;
    private image: string;

    constructor(book: Book) {
        this.id = book.id;
        this.isbn = book.isbn;
        this.category = book.category;
        this.title = book.title;
        this.author = book.author;
        this.language = book.language;
        this.publicationYear = book.publicationYear;
        this.publisher = book.publisher;
        this.image = book.image;
    }
}

File: spring-backend/src/main/java/library/app/backend/book/BookDto.java
package library.app.backend.book;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import library.app.backend.category.Category;
import lombok.AllArgsConstructor;
import org.springframework.hateoas.RepresentationModel;

import java.io.Serializable;
import java.util.List;
import java.util.UUID;

/**
 * DTO for {@link Book}
 */
@JsonIgnoreProperties(ignoreUnknown = true)
@AllArgsConstructor
public class BookDto extends RepresentationModel<BookDto> implements Serializable {
        UUID id;
        String isbn;
        List<Category> category;
        String title;
        String author;
        String language;
        Integer publicationYear;
        String publisher;
        String image;
        Byte isAvailable;

    public BookDto toBookDto(Book book) {
        return new BookDto(
                book.getId(),
                book.getIsbn(),
                book.getCategory(),
                book.getTitle(),
                book.getAuthor(),
                book.getLanguage(),
                book.getPublicationYear(),
                book.getPublisher(),
                book.getImage(),
                book.getIsAvailable()
        );
    }
}